### ISSUE TITLE
Custom crop frame

### ISSUE THREAD

**Issue by Talgat777** [albinmathew/PhotoCrop#4]
Hi, 

Is it possible to add the ability to set the custom crop frame or based on ratio? It looks like currently the lib supports only the square mode.

**Comment by albinmathew** [comment]
Yes its possible to add custom crop areas, currently the app supports square and circular cropping. I am working on custom ratio crop feature

**Comment by Talgat777** [comment]
When are you planing to release a new version with this feature?

**Comment by albinmathew** [comment]
Can you please explain the your actual requirement, so that I may suggest some workaround till new feature is implemented

**Comment by Talgat777** [comment]
I need to crop images with 4:3 ratio.

**Comment by rohanmahale** [comment]
Hi Albin. Any updates on the custom crop view frame?

**Comment by albinmathew** [comment]
@rohanmahale , @Talgat777 I was busy with lot of works and couldn't finish it. I will suggest you a workaround
1.Fork the repository
2.Make changes to CropOverlay class on photocrop library. Current implementation is seeking fixed cropping width and height. So to make a custom crop frame, use your own width and  height and never forget to use a dp to pixel converter for making something similar to 4.3 ratio.
3. This work around can be used then to make cropping any area for your wish, I have myself done this prior to writing this workaround
4.Now add the module "photo-crop-library" to your project as a dependency.
5. If everything works make a pull request. :+1:

**Comment by VannesEventurers** [comment]
How to make the circle center for all devices. Now it is somewhat aligned top. Could you help on it?

**Comment by albinmathew** [comment]
Have you adjusted app:marginTop="50dp" See the read me

**Comment by Mokshi1709** [comment]
You have specifed the scaletype of the image to centre, but if i change it to fitXY it shows error. Also, i want my requirement something similar to Instagram like cropping. the image need to be filled full background. Any help on this?

**Comment by Mokshi1709** [comment]
One more thing: how to get the circle cropped image alone finally.

**Comment by Vierco** [comment]
Hi @albinmathew 
Thanks for sharing this library
I'm also trying to change the ratio of cropping the image, but unfortunately, without success.
I read in your previous answer: "I have myself done prior to writing esta esta workaround"
Can you tell me what values ​​you have modified the library?
I tried changing many of them, for example:

CropOverlayView.java
private int cropHeight = DEFAULT_CROPWIDTH;
private int cropWidth = DEFAULT_CROPWIDTH;

but no appreciable changes

Thanks for your help
